src/neo/Ledger/Blockchain.cs

```diff
@@ -430,7 +432,8 @@ private VerifyResult OnNewTransaction(Transaction transaction)
 
         private void OnPersistCompleted(Block block)
         {
-            block_cache.Remove(block.Hash);
+            if (lastBlockhash != null) block_cache.Remove(lastBlockhash);
+            lastBlockhash = block.Hash;
```

Comment:
I think we need a FIFOCache.

Reply:
Or use HashSetCache https://github.com/neo-project/neo/blob/3e2c4dccb1e46a158a648dee36a761b8195ae838/src/neo/IO/Caching/HashSetCache.cs#L29-L31

Reply:
Why need cache multiple blocks?